library(RUVSeq)
reads_counts <- read.table("WT-merge.txt",row.names = 1,header = T,sep="\t",quote="")
head(reads_counts)
filter <- apply(reads_counts, 1, function(x) length(x[x>5])>=3)
head(filter)
filtered <- reads_counts[filter,]
head(filtered)
filtered=reads_counts
genes <- rownames(filtered)[grep("^ERCC", rownames(filtered),invert =T)]
head(genes)
spikes <- rownames(filtered)[grep("^ERCC", rownames(filtered))]
head(spikes)
x <- as.factor(rep(c("Ctl", "Trt"), each=6))
set <- newSeqExpressionSet(as.matrix(filtered),
phenoData = data.frame(x, row.names=colnames(filtered)))
set
library(RColorBrewer)
colors <- brewer.pal(3, "Set2")
par(mfrow = c(1,2))
pdf("RLE.PCA.pdf")
plotRLE(set, outline=FALSE, ylim=c(-4, 4), col=colors[x])
plotPCA(set, col=colors[x], cex=1.2)
#barplot(as.matrix(pData(set)), beside=TRUE)
dev.off()
seqRUVg <- RUVg(set, spikes, k=6)
pData(seqRUVg)
par(mfrow = c(1,2))
pdf("RLE.PCA.normlise.pdf")
plotRLE(seqRUVg, outline=FALSE, ylim=c(-4, 4), col=colors[x])
plotPCA(seqRUVg, col=colors[x], cex=1.2)
#barplot(as.matrix(pData(seqRUVg)), beside=TRUE)
dev.off()
head(normCounts(seqRUVg))
norm_Counts_matrix <- data.frame(normCounts(seqRUVg))
row.names(norm_Counts_matrix) <- row.names(set)
write.table(norm_Counts_matrix,"norm_Counts_matrix.xls",quote=F,row.names=T,sep="\t")

library(edgeR)
library(ggplot2)
library(GenomicFeatures)
txdb <- makeTxDbFromGFF("/home/l/backup1/refgenome/Arabidopsis/ERCC92/Arabidopsis_ERCC.gtf",format="gtf")
exons_gene <- exonsBy(txdb, by = "gene")
exons_gene_lens <- lapply(exons_gene,function(x){sum(width(reduce(x)))})
head(exons_gene_lens)
write.table(exons_gene_lens,"exons_gene_lens.xls",sep="\t",quote=F,row.names=F)
exons_gene_lens <- read.csv("exons_gene_lens.xls",stringsAsFactors = F,header=F,sep="\t")
texons_gene_lens <- t(exons_gene_lens)
head(texons_gene_lens)
write.table(texons_gene_lens,"texons_gene_lens.xls",sep="\t",quote=F,row.names=F)
texons_gene_lens <- read.table("texons_gene_lens.xls",row.names=1,header=T,sep="\t")
texons_gene_lens <- as.matrix(texons_gene_lens)
names(texons_gene_lens) <- row.names(texons_gene_lens)
reads_matrix <- normCounts(seqRUVg)
row.names(reads_matrix) <- row.names(set)
rownames(reads_matrix)[!rownames(reads_matrix) %in% names(texons_gene_lens)]
d <- DGEList(counts = reads_matrix)
d$genes$Length <- c(texons_gene_lens)
mycounts <- rpkm(d)
write.table(mycounts,"total_readscounts.RPKM.txt",quote=F,row.names=T)

